### ISSUE TITLE
New cop: Style/ReturnNilCondition

### ISSUE THREAD

**Issue by Ana06** [rubocop-hq/rubocop#6033]
Introduce a new cop, which could be called `Style/ReturnNilCondition`, to avoid only returning `nil` in a condition branch.

# Examples

## bad

```ruby
user.present? ? user.name : nil

if user.present?
  user.name
else
  nil
end

emails.empty? nil : emails.count

if emails.empty?
  nil
else
  emails.count
end
```

## good

```ruby
user.name if user.present?

emails.count unless emails.empty?
```

**Comment by Ana06** [comment]
I think there is not a cop which complains about that already :thinking:

**Comment by Edouard-chin** [comment]
FYI there was some interesting discussion [here](https://github.com/rails/rails/pull/32845) on using `else nil` which is sometimes preferred and intentional.

Not saying that we shouldn't do this cop but just wanted to give some context

**Comment by dvandersluis** [comment]
FYI `Style/EmptyElse` will catch 

```ruby
if user.present?
  user.name
else
  nil
end
```

but it doesn't look like anything catches any of the other examples.

**Comment by bbatsov** [comment]
That's a great cop idea! Returning `nil` has ruined too many apis. :-) 

@dvandersluis Yeah, the point of the other cop is to just to encourage/discourage implicit `else` branches. It's not related to what's suggested here.

**Comment by Drenmi** [comment]
> That's a great cop idea! Returning nil has ruined too many apis.

If I understand correctly, this won't disallow returning `nil`, just making the return value implicit. 😅

**Comment by bbatsov** [comment]
Ops, I misread the ticket then. Well, now I'm not that excited anymore. 🤣

**Comment by Ana06** [comment]
:sob:
